smarts/env/wrappers/single_agent.py

```diff
@@ -0,0 +1,72 @@
+# MIT License
+#
+# Copyright (C) 2021. Huawei Technologies Co., Ltd. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+from typing import Any, Tuple
+
+import gym
+
+
+class SingleAgent(gym.Wrapper):
```

Comment:
OK, I think we will want to use this explicitly in the [single_agent.py](https://github.com/huawei-noah/SMARTS/blob/wrapper-single-agent/examples/single_agent.py) example.

Reply:
Currently, `single_agent.py` illustrates the multi-agent interface of SMARTS. The `episode.record_step()` on line 67 works on dictionary inputs for multi-agents. 

I attempted to change `single_agent.py` to use `SingleAgent` wrapper. Using `SingleAgent` wrapper makes the code look like normal single-agent gym-compliant code. But line 67 will then be written as:
episode.record_step({AGENT_ID:obs}, {AGENT_ID:reward}, {AGENT_ID:done, “__all__”:done}, {AGENT_ID:info}) 

1) So should we proceed with using `SingleAgent` wrapper in `single_agent.py` example? 
2) If we are using `SingleAgent` wrapper in `single_agent.py`, shall we just remove the `episode.record_step()` functionality in order to streamline the example?

Reply:
> 1. Should we proceed with using SingleAgent wrapper in single_agent.py example?

Yes, I think we should still do so.

> 2. If we are using `SingleAgent` wrapper in `single_agent.py`, shall we just remove the` episode.record_step()` functionality in order to streamline the example?

The use of record step is just for visual feedback when running the example. I am not sure if we can get rid of it, I could suggest we can make an overload or simplified forwarding version of `record_step()`.

Reply:
Introduced `SingleAgent` wrapper into `single_agent.py` example.